### ISSUE TITLE
Lifecycle sidecar constantly CPU throttled

### ISSUE THREAD

**Issue by adinhodovic** [hashicorp/consul-helm#738]
### Overview of the Issue

CPU limits set too low for the consul-connect sidecar.

### Reproduction Steps

1. When running helm install with the default values and injecting the consul connect sidecar to a pod the CPU is consistently throttled and causing Prometheus alerts to be triggered. [The CPU throttling alert is taken from the `kubernetes-mixin` project.](https://github.com/kubernetes-monitoring/kubernetes-mixin/blob/master/alerts/resource_alerts.libsonnet#L149-L162)
1. View error
![image](https://user-images.githubusercontent.com/12065867/101904899-dd6e1d80-3bb6-11eb-9567-3040a7df2016.png)
Grafana graph:
![image](https://user-images.githubusercontent.com/12065867/101906094-b7e21380-3bb8-11eb-9107-4c3e21340bb2.png)

The above image goes for every pod that have injected the sidecar.

I assume this is due to that the sidecar is CPU intensive in short spans, exhausting CPU time period for that short span.

### Expected behavior

What was the expected result?

No alerts and no CPU exhaustion. I would assume performance is degraded as well, have not looked into it in depth yet. Currently I overwrite the CPU limit, but would prefer better defaults in the values file and also to be have the option to disable CPU limits completely(don't think I can atm, how do I set null just for the CPU limit?).
### Environment details

If not already included, please provide the following:
- `consul-k8s` version: 1.9
- `consul-helm` version: 0.27.0
- `values.yaml` used to deploy the helm chart: None specific

**Comment by ashwin-venkatesh** [comment]
Hey @adinhodovic. Thanks for filing the report. What CPU limits/requests did you set that saw this problem mitigated? We can certainly look into adjusting the default limits and requests on our end. 
> have the option to disable CPU limits completely
AFAIR explicitly setting the value of the limit to `0` should have the desired effect.

**Comment by adinhodovic** [comment]
Hi @ashwin-venkatesh,

> Thanks for filing the report. What CPU limits/requests did you set that saw this problem mitigated? We can certainly look into adjusting the default limits and requests on our end.

I tend to avoid to set limits on CPU on very spiky workloads. I pushed the limits to half a CPU before the throttling alerts went a way and that is quite high. I assume the sync that happens every 10 seconds and it is very cpu heavy for a very short time if my understanding is correct.

> AFAIR explicitly setting the value of the limit to 0 should have the desired effect.
I think this sets the limit to 0 CPUs and Kubernetes will generate a weird error `Invalid value: "20m": must be less than or equal to cpu limit`.

With the settings on the pod:

```yaml
      limits:
        cpu: 0
        memory: 50Mi
      requests:
        cpu: 20m
        memory: 25Mi
```

And the lifecycle-webhook-deployment:

```
            -lifecycle-sidecar-memory-limit=50Mi \
            -lifecycle-sidecar-memory-request=25Mi \
            -lifecycle-sidecar-cpu-limit=0 \
            -lifecycle-sidecar-cpu-request=20m \
```

And the helm value:

```yaml
    global:
      lifecycleSidecarContainer:
        resources:
          limits:
            cpu: "0"
```
I would prefer to not set it completely.

**Comment by ashwin-venkatesh** [comment]
@adinhodovic 😅 I set both the limit and the request to `0` when I tested it which worked out. Unfortunately, I don't think it can be completely unset given our values file will default it if that block isn't set.

The sync does happen every 10 seconds. We are looking at re-evaluating our implementation of the sync so that it is less CPU intensive, but it could take a while before it gets done unfortunately.

**Comment by adinhodovic** [comment]
> @adinhodovic sweat_smile I set both the limit and the request to `0` when I tested it which worked out. Unfortunately, I don't think it can be completely unset given our values file will default it if that block isn't set.
> 
> The sync does happen every 10 seconds. We are looking at re-evaluating our implementation of the sync so that it is less CPU intensive, but it could take a while before it gets done unfortunately.

Fair enough, however I would like to only set requests and remove the limits. Does that work for you? If requests are not set then e.g HPA scaling won't work since all containers need requests set within a pod :shrug:.

**Comment by lkysow** [comment]
Closed by #457 that removes need for lifecycle sidecar re-registration. Will be in next release.
